### PR TITLE
Handle OAuth Flow and Add Functionality to Invalidate an HttpSession

### PR DESCRIPTION
**Pull request by cxfloeder**
Separate the Login and OAuth functionality into separate servlets. The OAuthConsentServlet guides users through the process to get access tokens to use the Sheets API. This servlet will be called after the UI sees that the user is logged in. The InvalidateSessionServlet ends a session so that access tokens aren't preserved when the app is exited.

TODO: Structure UI in a way that properly uses the Login and OAuth servlets.

### CODE REVIEW

File: src/main/java/servlets/OAuthCallbackServlet.java [hunk 1]
```diff
@@ -45,20 +45,25 @@
...
+
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String error = request.getParameter(ERROR_PARAM);
     String authCode = request.getParameter(CODE_PARAM);
     String state = request.getParameter(STATE_PARAM);
 
+    System.out.println("GOT HERE");
+
```
Comment (cadecrandall) [inline src/main/java/servlets/OAuthCallbackServlet.java:58]: can you remove this before merging?

Comment (cxfloeder) [reply src/main/java/servlets/OAuthCallbackServlet.java:58]: Yes I can.

File: src/main/java/servlets/BillingConfig.java [hunk 2]
```diff
@@ -66,7 +66,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
      * TODO: @cade, eventually you will want to pass this access token into the 
      * updateSheets() method in the SheetsConverter class.
      */
-    HttpSession session = request.getSession();
+    HttpSession session = request.getSession(false);
```
Comment (ianjjohnson) [inline src/main/java/servlets/BillingConfig.java:69]: please document what this parameter means

File: src/main/java/servlets/InvalidateSessionServlet.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import com.google.appengine.api.users.UserServiceFactory;
+import util.UserAuthUtil;
+
+@WebServlet("/Invalidate")
+public class InvalidateSessionServlet extends HttpServlet {
+
+  // Invalidate the HttpSession that holds the access token.
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    HttpSession session = request.getSession(false);
```
Comment (ianjjohnson) [inline src/main/java/servlets/InvalidateSessionServlet.java:29]: same here

File: src/main/java/servlets/InvalidateSessionServlet.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+package servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.*;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import util.UserAuthUtil;
+
+@WebServlet("/Invalidate")
```
Comment (ianjjohnson) [inline src/main/java/servlets/InvalidateSessionServlet.java:23]: Should this be "InvalidateSession"?

Comment (cxfloeder) [reply src/main/java/servlets/InvalidateSessionServlet.java:23]: I could switch it to that, I just thought only having Invalidate would make typing it easier.

File: src/main/java/servlets/BillingConfig.java (hunk 2 above)
Comment (cxfloeder) [reply src/main/java/servlets/BillingConfig.java:69]: getSession(false) will only get a session if one exists. getSession(true) will create a session if none currently exist.

**Pull request merged**